core/src/test/scala/org/broadinstitute/dsde/rawls/entities/local/CaseSensitivitySpec.scala

```diff
@@ -609,115 +655,93 @@ class CaseSensitivitySpec extends AnyFreeSpec with Matchers with TestDriverCompo
-        val queriedAttributes = queriedEntities.results.head.attributes
+        queriedEntities.size shouldBe 1
+        val queriedAttributes = queriedEntities.head.attributes
         exemplarAttributeNames.size should equal(queriedAttributes.size)
         queriedAttributes.keys.map(toDelimitedName) should contain theSameElementsAs exemplarAttributeNames
       }
 
-      "should delete all associated attributes when deleting entities of any type" in withTestDataServices { _ =>
-        // save case insensitive attribute data
-        runAndWait(entityQuery.save(testWorkspace.workspace, caseInsensitiveAttributeData))
-
-        // get provider
-        val provider = new LocalEntityProvider(EntityRequestArguments(testWorkspace.workspace, testContext),
-                                               slickDataSource,
-                                               true,
-                                               testConf.getDuration("entities.queryTimeout"),
-                                               "metricsBaseName"
-        )
-
-        // delete our test entity
-        provider.deleteEntities(Seq(EntityPointer("cat", "005")), testContext).futureValue shouldBe 1
-
-        // make sure all attributes are marked as deleted
-        import driver.api._
-
-        val allAttributes = runAndWait(entityAttributeShardQuery(testWorkspace.workspace).result)
-        exemplarAttributeNames.size shouldBe allAttributes.size
-        allAttributes.foreach(attr => assert(attr.deleted))
-      }
-
       "should create all attributes for a new entity" in withTestDataServices { _ =>
         // get provider
-        val provider = new LocalEntityProvider(EntityRequestArguments(testWorkspace.workspace, testContext),
-                                               slickDataSource,
-                                               true,
-                                               testConf.getDuration("entities.queryTimeout"),
-                                               "metricsBaseName"
-        )
+        val provider = defaultProvider
 
         // create our entity
         provider.createEntity(caseInsensitiveAttributeData.head, testContext).futureValue // Entity
 
-        // make sure all attributes are created
-        import driver.api._
-
-        val allAttributes = runAndWait(entityAttributeShardQuery(testWorkspace.workspace).result)
-        exemplarAttributeNames.size shouldBe allAttributes.size
-        allAttributes.map(attr =>
-          toDelimitedName(AttributeName(attr.namespace, attr.name))
-        ) should contain theSameElementsAs exemplarAttributeNames
+        val allAttributes = provider
+          .getEntity(caseInsensitiveAttributeData.head.entityType, caseInsensitiveAttributeData.head.name, testContext)
+          .futureValue
+          .attributes
+        allAttributes.keys.map(AttributeName.toDelimitedName) should contain theSameElementsAs exemplarAttributeNames
```

Comment:
I removed the assertion on `.size`. The `should contain theSameElementsAs` assertion covers that, and outputs more helpful errors when the test fails.